tests/tests_unit/test_api/test_data_modeling/test_containers.py

```diff
@@ -0,0 +1,95 @@
+import re
+from typing import Any
+
+import pytest
+
+from cognite.client import CogniteClient
+from cognite.client.data_classes.data_modeling import ContainerApply, ContainerId, ContainerProperty, Text
+from cognite.client.data_classes.data_modeling.containers import BTreeIndex, RequiresConstraint
+
+EXAMPLE_CONTAINER = {
+    "space": "testspace",
+    "externalId": "container1",
+    "name": "string",
+    "description": "string",
+    "usedFor": "node",
+    "properties": {
+        "prop1": {
+            "immutable": False,
+            "nullable": True,
+            "autoIncrement": False,
+            "defaultValue": "string",
+            "description": "string",
+            "name": "string",
+            "type": {"type": "text", "list": False, "collation": "ucs_basic"},
+        }
+    },
+    "constraints": {
+        "constraint1": {
+            "constraintType": "requires",
+            "require": {"type": "container", "space": "string", "externalId": "string"},
+        }
+    },
+    "indexes": {"index1": {"properties": ["prop1"], "indexType": "btree", "cursorable": False, "bySpace": False}},
+    "createdTime": 123,
+    "lastUpdatedTime": 123,
+    "isGlobal": True,
+}
+
+
+@pytest.fixture
+def mock_containers_response(rsps: Any, cognite_client: CogniteClient):
+    response_body = {"items": [EXAMPLE_CONTAINER]}
+    url_pattern = re.compile(
+        re.escape(cognite_client.data_modeling.containers._get_base_url_with_base_path())
+        + "/models/containers(/byids)?$"
+    )
+    rsps.add(rsps.POST, url_pattern, status=200, json=response_body)
+    yield rsps
+
+
+@pytest.fixture
+def mock_delete_index_response(rsps: Any, cognite_client: CogniteClient):
+    response_body = {
+        "items": [
+            {
+                "space": EXAMPLE_CONTAINER["space"],
+                "containerExternalId": EXAMPLE_CONTAINER["externalId"],
+                "identifier": "index1",
+            }
+        ]
+    }
+    url_pattern = re.compile(
+        re.escape(cognite_client.data_modeling.containers._get_base_url_with_base_path())
+        + "/models/containers/indexes/delete$"
+    )
+    rsps.add(rsps.POST, url_pattern, status=200, json=response_body)
+    yield rsps
+
+
+class TestContainersApi:
+    def test_apply_retrieve_and_delete_index(
+        self, cognite_client: CogniteClient, mock_containers_response: Any, mock_delete_index_response: Any
+    ) -> None:
+        new_container = ContainerApply(
```

Comment:
I am almost in favor of removing this test. I think by making it an unit test we only testing that the serialization is correct, which should be tested on the data class, and that we call the correct endpoint in the API method. If there are any changes to the endpoint, it would give a false sense of security.

Reply:
It's not the python sdk tests' responsibility to ensure that the service is working correctly. We have smoke tests for all services. The pysdk tests should ensure that we correctly integrate with the API. Any changes to the endpoint should be backwards compatible.